back/spec/acceptance/areas_spec.rb

```
@@ -136,4 +136,59 @@
       end
     end
   end
+
+  context 'when citizen' do
+    before do
+      user_header_token
+
+      @areas = create_list(:area, 4)
+
+      @projects = %w[published archived published]
+                  .map { |ps| create(:project, admin_publication_attributes: { publication_status: ps }) }
+
+      @projects.each_with_index do |project, i|
+        project.areas << @areas[i]
+        project.save!
+      end
+    end
+
+    get 'web_api/v1/areas' do
+      with_options scope: :page do
+        parameter :number, 'Page number'
+        parameter :size, 'Number of areas per page'
+      end
+      parameter :for_homepage_filter, 'Filter: areas of only visible non-draft projects not in draft folders', required: false
+
+      example 'List only selected areas does not include areas only used by draft projects' do
+        @projects[0].update(admin_publication_attributes: { publication_status: 'draft' })
```

Use update! instead of update to avoid silent failures.